lib/Ed25519KeyPair.js

```diff
@@ -42,6 +42,9 @@ class Ed25519KeyPair extends LDKeyPair {
     this.type = 'Ed25519VerificationKey2018';
     this.privateKeyBase58 = options.privateKeyBase58;
     this.publicKeyBase58 = options.publicKeyBase58;
+    if(this.controller && this.publicKeyBase58 && !this.id) {
+      this.id = `${this.controller}#${this.fingerprint()}`;
```

Comment:
I note that even prior to this patch there is no validation that options.publicKeyBase58 has been specified.  Now with this, we'll get some unusual error when the fingerprint function tries to compute the fingerprint.  Would it be possible to add some validation on `publicKeyBase58`?

Reply:
@mattcollier hehe, I think I was fixing that right as you were typing the comment.

Reply:
@mattcollier unless you mean validating it in the `fingerprint()` method itself?

Reply:
Well, this is one way to go about it.  But `publicKeyBae58` is a required property right?  If we just validate it as such, then we don't need the extra conditional.

The docs (if accurate) indicate that both public and private key material are required, but I note in your test you're only providing publicKey material.

https://github.com/digitalbazaar/crypto-ld/blob/master/docs/Ed25519KeyPair.md

Reply:
@mattcollier hmm, good point. So for one, I should fix the docs - private key is definitely optional.
And I'll add a general validation of the public key in the constructor.

Reply:
Done.

Reply:
Great!  Thanks, the same concept applies to RSA as well right?

Reply:
Good question. I'm honestly not sure what is going on with the RSA fingerprint code currently. (I looked at the constructor, and it _appears_ the public key is optional? I am open to suggestions.)